service-base/src/main/java/org/eclipse/hono/service/metric/Metrics.java

```diff
@@ -13,11 +13,132 @@
 
 package org.eclipse.hono.service.metric;
 
+import org.eclipse.hono.util.EndpointType;
+
+import io.micrometer.core.instrument.Timer.Sample;
+
 /**
  * A collector for metrics.
  */
 public interface Metrics {
 
+    /**
+     * The direction of a message.
+     *
+     */
+    enum Direction {
```

Comment:
I would propose to use the convention of all uppercase enum literals.

Reply:
Yes, I thought so too

Reply:
done